lib/fitgem/client.rb

```diff
@@ -249,9 +166,17 @@ def raw_delete(path, headers={})
       def extract_response_body(response)
         return {} if response.nil?
 
-        raise ServiceUnavailableError if response.code == '503'
+        raise ServiceUnavailableError if response.status == 503
 
         response.body.nil? ? {} : JSON.parse(response.body)
```

Comment:
In some live testing I did a couple days ago I was seeing the Fitbit API returning an empty string in the body, not `nil`. This was causing this line to attempt to `JSON.parse('')` which raises `JSON::ParserError: A JSON text must at least contain two octets!`